gtest_parallel_tests.py

```diff
@@ -173,8 +167,40 @@ def test_run_task_basic(self):
         sorted(task.task_id for task in task_manager.failed),
         sorted(task.task_id for task in task_mock_factory.failed))
 
-    self.assertEqual(task_manager.global_exit_code, 1)
+    self.assertEqual(task_manager.global_exit_code, expected_exit_code)
+
+  def test_passing_task_succeeds(self):
+    self.execute_tasks(tasks=[self.passing_task], retries=0,
+                       expected_exit_code=0)
+
+  def test_failing_task_fails(self):
+    self.execute_tasks(tasks=[self.failing_task], retries=0,
+                       expected_exit_code=1)
+
+  def test_failing_task_fails_even_with_retries(self):
+    self.execute_tasks(tasks=[self.failing_task], retries=2,
+                       expected_exit_code=3)
+
+  def test_executing_passing_and_failing_fails(self):
+    self.execute_tasks(tasks=[self.failing_task, self.passing_task],
+                       retries=2, expected_exit_code=3)
+
+    self.execute_tasks(tasks=[self.passing_task, self.failing_task],
+                       retries=2, expected_exit_code=3)
+
+  def test_task_that_fails_once_succeeds_with_two_attempts(self):
+    self.execute_tasks(tasks=[self.fails_once_then_succeeds],
+                       retries=1, expected_exit_code=0)
+
+  def test_task_that_fails_twice_fails_with_two_attempts(self):
```

Comment:
Or test_task_gets_repeated_according_to_retries(

Your call. Do add a comment that says # Executes test and retries once, not enough for the test to start passing, so gtest-parallel should return an error.

Add similar comments where you think they are necessary.